src/Common/CaresPTRResolver.cpp

```diff
@@ -128,7 +128,7 @@ namespace DB
             int number_of_fds_ready = 0;
             if (!readable_sockets.empty())
             {
-                number_of_fds_ready = poll(readable_sockets.data(), readable_sockets.size(), timeout);
+                number_of_fds_ready = poll(readable_sockets.data(), static_cast<unsigned int>(readable_sockets.size()), static_cast<int>(timeout));
```

Comment:
Seems that `nfds_t` is `unsigned long int` (second argument), is it `unsigned int` for you?

Reply:
I see, it is. Let's cast to `nfds_t` then (second argument).

```suggestion
                number_of_fds_ready = poll(readable_sockets.data(), static_cast<nfds_t int>(readable_sockets.size()), static_cast<int>(timeout));
```

Reply:
Can you try to rebase the branch? (So that CI sees labels)